docs/docsite/rst/dev_guide/developing_plugins.rst

```diff
@@ -40,16 +40,20 @@ Any strings returned by your plugin that could ever contain non-ASCII characters
     result_string = to_text(result_string)
 
 Plugin Configuration
-````````````````````
+^^^^^^^^^^^^^^^^^^^^
 
 Starting with Ansible version 2.4, we are unifying how each plugin type is configured and how they get those settings.  Plugins will be able to declare their requirements and have Ansible provide them with a resolved'configuration. Starting with Ansible 2.4 both callback and connection type plugins can use this system.
 
-Most plugins will be able to use  ``self._options[<optionname>]`` to access the settings, except callbacks that use ``self._plugin_options[<optionname>]``.
+Most plugins will be able to use  ``self.get_option(<optionname>)`` to access the settings.
+These are pre-populated by a ``self.set_options()`` call, for most plugin types this is done by the controller,
+but for some types you might need to do this explicitly.
+Of course, if you don't have any configurable options, you can ignore this.
 
-Plugins that support embedded documentation (see `ansible-doc` for the list) are now required to provide well-formed doc strings to be considered for merge into the Ansible repo.
+Plugins that support embedded documentation (see :ref:`ansible-doc` for the list) are now required to provide well-formed doc strings to be considered for merge into the Ansible repo. This documentations also doubles as 'configuration definition' so they will never be out of sync.
```

Comment:
Instead of "are now required to provide" I think "must now include" is clearer.

Typo - s on "documentations" should be removed.

Heh, can we promise that anything will "never be out of sync"?

Reply:
In this case, yes. Since there is nothing to 'sync' it cannot 'fall out of sync'